All props in stateful storage should be non-AD

lindsayad opened the pull request:
Otherwise we will multiply our memory usage by 51x which isn't ideal.

Closes #16836

lindsayad committed:
All props in stateful storage should be non-AD

Otherwise we will multiply our memory usage by 51x which isn't ideal.

Closes #16836

hugary1995 commented:
I'll re-do some of the profiling in #16466 when this gets merged. The profiling data told me the bottleneck of the AD performance there was related to memory allocation.

lindsayad commented:
Copying from #16773:

Before this PR, with a finite strain input from @makeclean, I had this graph:
![fixed-temp](https://user-images.githubusercontent.com/10248304/106531103-6eb88c80-64a2-11eb-9bc8-2f6a94f157c0.png)
which was taken from the middle of initing stateful props, and shortly before I canceled the simulation so my computer wouldn't run out of memory to this graph:
![post-stateful-fixes](https://user-images.githubusercontent.com/10248304/106531166-8bed5b00-64a2-11eb-9e0e-e304a340e007.png)
which was taken *after* initing stateful properties...because now we can get to that point without running out of memory. So 84 GB for stateful properties (and that number was still growing) to 4GB. I'm a big fan. Thanks for your helpful inputs @makeclean. Good news @dschwen @recuero !

dschwen commented:
Nice!

> Otherwise we will multiply our memory usage by 51x which isn't ideal

Or a lot more. We're already running Bison with AD vector sizes > 100 for 2nd order elements in 3D...

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/671041/) on 7f615ee wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16845/site)

This comment will be updated on new commits.

lindsayad committed:
Remove now extraneous shift code

The properties in `MaterialPropertyStorage` are all regular
`MaterialProperty` so there is no need to change property types when
shifting anymore.

Refs #16836
